File: src/main/java/com/lannstark/lec12/Lec12Main.java
package com.lannstark.lec12;

public class Lec12Main {

  public static void main(String[] args) {
    // Person.Factory.newBaby("ABC");
    
    moveSomething(new Movable() {
      @Override
      public void move() {
        System.out.println("움직인다!!");
      }

      @Override
      public void fly() {
        System.out.println("난다!!");
      }
    });
    
  }

  public static void moveSomething(Movable moveable) {
    moveable.move();
    moveable.fly();
  }

}

File: src/main/java/com/lannstark/lec12/Lec12Main.kt
package com.lannstark.lec12

fun main() {
    moveSomething(object : Movable {
        override fun move() {
            println("움직인다!!!!!")
        }

        override fun fly() {
            println("난다!!!!!")
        }
    })
}

private fun moveSomething(moveable: Movable) {
    moveable.move()
    moveable.fly()
}